src/client/pages/Afis/useAfisThemaData.hook.tsx

```diff
@@ -0,0 +1,189 @@
+import { ReactNode, useEffect, useMemo } from 'react';
+import {
+  AfisBusinessPartnerDetailsTransformed,
+  AfisBusinessPartnerKnownResponse,
+  AfisFactuur,
+} from '../../../server/services/afis/afis-types';
+import {
+  hasFailedDependency,
+  isError,
+  isLoading,
+} from '../../../universal/helpers/api';
+import { DisplayProps } from '../../components/Table/TableV2';
+import { BFFApiUrls } from '../../config/api';
+import { BagThemas } from '../../config/thema';
+import { useAppStateBagApi, useAppStateGetter } from '../../hooks/useAppState';
+import {
+  eMandateTableConfig,
+  facturenTableConfig,
+  ListPageParamKind,
+  listPageParamKind,
+  listPageTitle,
+  routes,
+} from './Afis-thema-config';
+import { MaRouterLink } from '../../components/MaLink/MaLink';
+
+import { DocumentLink } from '../../components/DocumentList/DocumentLink';
+import { capitalizeFirstLetter } from '../../../universal/helpers/text';
+import { GenericDocument } from '../../../universal/types';
+
+export function useAfisThemaData(kind?: ListPageParamKind) {
+  const { AFIS } = useAppStateGetter();
+  const businessPartnerIdEncrypted = AFIS.content?.businessPartnerIdEncrypted;
+  const [facturen, facturenApi, fetchFacturen] = useAppStateBagApi<
+    { [key in ListPageParamKind]: AfisFactuur[] } | AfisFactuur[] | null
+  >({
+    bagThema: kind ? BagThemas.AFIS : BagThemas.AFIS_OVERZICHT,
+    key: `${businessPartnerIdEncrypted}-facturen`,
+  });
+
+  useEffect(() => {
+    if (businessPartnerIdEncrypted) {
+      const url = kind
+        ? `${BFFApiUrls.AFIS_FACTUREN}/${kind}/${businessPartnerIdEncrypted}`
+        : `${BFFApiUrls.AFIS_FACTUREN_OVERZICHT}/${businessPartnerIdEncrypted}`;
+      fetchFacturen({ url });
+    }
+  }, [businessPartnerIdEncrypted, fetchFacturen, kind]);
+
+  const updatedFacturen = useMemo(() => {
+    return kind
+      ? Array.isArray(facturen)
+        ? (facturen as AfisFactuur[])?.map(mapFactuur)
+        : []
+      : facturen && Object.keys(facturen).length > 0
+        ? Object.fromEntries(
+            Object.entries(facturen).map(([key, value]) => [
+              key,
+              value.map(mapFactuur),
+            ])
+          )
+        : { open: [], closed: [] };
+  }, [facturen, kind]);
+
+  return {
+    businessPartnerIdEncrypted:
+      AFIS.content?.businessPartnerIdEncrypted ?? null,
+    isThemaPaginaLoading: isLoading(AFIS),
+    isThemaPaginaError: isError(AFIS, false),
+    routes,
+    facturenTableConfig,
+    listPageTitle,
+    listPageParamKind,
+    isFacturenError: facturenApi.isError,
+    isFacturenLoading: facturenApi.isLoading,
+    facturen: updatedFacturen,
+  };
+}
+
+function mapFactuur(factuur: AfisFactuur) {
+  return {
+    ...factuur,
+    status: getInvoiceStatusDescription(factuur),
+    factuurNummer: factuur.documentDownloadLink ? (
```

Comment:
Het nadeel van overschrijven van props is dat je de originele waardes niet meer kan gebruiken. Je kunt er ook voor kiezen een additional type te maken:

```
type AfisFactuurFrontend = AfisFactuur & {
  statusDescription: ..,
  factuurNummerEL: ...
}